src/main/scala/com.snowplowanalytics.snowplow.badrows/BadRow.scala

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2018-2019 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.snowplow.badrows
+
+import io.circe.{Encoder, Decoder}
+import io.circe.generic.semiauto._
+import io.circe.syntax._
+import io.circe.Json
+
+import cats.syntax.functor._
+
+import com.snowplowanalytics.iglu.core.{SelfDescribingData, SchemaKey}
+import com.snowplowanalytics.iglu.core.circe.instances._
+
+sealed trait BadRow {
+  def schemaKey: SchemaKey
+  def selfDescribinData: SelfDescribingData[Json] = SelfDescribingData(schemaKey, this.asJson)
+  def compact: String = selfDescribinData.asJson.noSpaces
+}
+
+object BadRow {
+  implicit val badRowEncoder: Encoder[BadRow] = Encoder.instance {
+    // Stream Collector
```

Comment:
What about: https://github.com/snowplow/snowplow/blob/master/2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala#L283 ?

Reply:
It seems I messed up the rebasing: https://github.com/snowplow/snowplow/pull/4011/files#diff-4d9d5134bd9a8a3df816fec651847ad4R277

Tests for SSC still don't compile due rebasing, but I hope I would catch it. But great point - we proably need to add another bad row type?

Reply:
Apparently, it wasn't my fault - we just decided to drop it https://github.com/snowplow-incubator/snowplow-badrows/issues/24